python/sglang/srt/function_call_parser.py

```diff
@@ -351,15 +358,24 @@ def detect_and_parse(self, text: str, tools: List[Tool]) -> StreamingParseResult
         pattern = rf"{self.bot_token}(.*?){self.eot_token}"
         match_result_list = re.findall(pattern, text, re.DOTALL)
         calls = []
-        for match_result in match_result_list:
-            match_result = json.loads(match_result)
-            calls.extend(self.parse_base_json(match_result, tools))
-        return StreamingParseResult(normal_text=normal_text, calls=calls)
+        try:
+            for match_result in match_result_list:
+                try:
+                    parsed_result = json.loads(match_result)
+                    calls.extend(self.parse_base_json(parsed_result, tools))
```

Comment:
It seems the pattern here is in each Detector that requires json function format, we first tries to load the result, then pass it to `BaseFormatDetector.parser_base_json`. 
`json_str → json.loads() → Python objects → json.dumps() → json_str`

This is quite in-efficient IMO. And causes this issue with double dumping at the first place.

Why don't we simply pass the match_result to it without `json.loads`? 

In this way:
1. We call json.loads inside `BaseFormatDetector.parser_base_json`, easily avoid this json double dumping.
2. We do try-catch once there, without the need of try-catch in each Detector.